src/App.tsx

```diff
@@ -7,63 +8,91 @@ import { useEffect, useState } from 'react';
+  const [room, setRoom] = useState(dummyState)
 
-  db.collection('rooms').doc('1').set({
-    users: {
-      0: {
-        id: '1214142414',
-        name: 'nick',
-        stack: 100,
-        action: '30',
-        cards: ['10s', '8d'],
-        seat: 1,
-      },
-      1: {
-        id: '1248794817',
-        name: 'justin',
-        stack: 50,
-        action: 'fold',
-        cards: ['Ks', 'Qd'],
-        seat: 0,
-      }
-    },
-    pot: "100",
-    board: ['Ks', 'As', '4c'],
-    button: 0,
-    blindSize: .02,
-    buyIn: 10,
-    actor: 0,
+useEffect( () => {
+  docRef.onSnapshot((doc) => {
+    if(doc.exists) {
+      console.log('Firebase data:', doc.data());
+      let data = doc.data();
+      if(data) {setRoom({
+        deck: data.deck,
+        board: data.board,
+        button: data.button,
+        blindSize: data.blindSize,
+        users: data.users,
+        pot: data.pot,
+        buyIn: data.buyIn,
+        actor: data.actor
+      })};
+    } else {
+      console.log('No such document!');
+    }
   })
-  .then(function() {
-    console.log("Document successfully written!");
-  })
-  .catch(function(error) {
-      console.error("Error writing document: ", error);
-  });
+}, []);
+
 
+  const dealCard = (position) => {
+    if (position === 'board') {
+      return docRef.update({
+        board: firebase.firestore.FieldValue.arrayUnion(room.deck[0]),
+        deck: firebase.firestore.FieldValue.arrayRemove(room.deck[0])
+      })
+      .then(() => console.log('board updated'))
+      .catch((err) => console.log(err));
+    } else {
+      let playerHand = `users.${position}.cards`
+      return docRef.update({
+        [playerHand]: firebase.firestore.FieldValue.arrayUnion(room.deck[0]),
+        deck: firebase.firestore.FieldValue.arrayRemove(room.deck[0])
+      })
+      .then(() => console.log('player updated:', room.users[0]))
+      .catch((err) => console.log(err));
+    }
+  };
+
+  const clearHand = (position) => {
+    let playerHand = `users.${position}.cards`
+    return docRef.update({
+      [playerHand]: []
+    });
+  };
 
-  const defHandStyle = {
-    maxHeight:'20vh',
-    padding: 0,
+  const clearCards = () => {
+    
+    return docRef.update({
+      board: [],
+      deck: deck52
+    })
+    .then(() => console.log('board cleared'))
+    .catch(err => console.log(err));
   };
 
   return (
     <div className="App">
-      <header className="App-header">
-        <Hand cards={[
-          { rank: 1, suit: 0 },
-          { rank: 11, suit: 2 },
-          ]} hidden={false} style={defHandStyle}
-        />
-        <p>
-        {displayText}
-        </p>
-      </header>
+      <Board boardCards={room.board} clearCards = {clearCards} dealCard={() => dealCard('board')} pot={room.pot} />
+      <PlayArea>
+        {Object.keys(room.users).map(key => <Seat hand={room.users[key].cards} dealCard={() => dealCard(key)} clearHand={() => clearHand(key)} player={room.users[key]} />)}
```

Comment:
Too much for one line. Make a line break after the opening paren for .map

use `Object.entries` to key both the key and value
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries